integration-libs/opf/base/components/opf-quick-buy/apple-pay/apple-pay.service.ts

```diff
@@ -41,44 +43,77 @@ export class ApplePayService {
   protected cartHandlerService = inject(OpfCartHandlerService);
 
   protected paymentInProgress = false;
-  protected localCart: LocalCart = {
-    quantity: 0,
+
+  protected initialTransactionDetails: ApplePayTransactionDetails = {
+    context: OpfQuickBuyLocation.PRODUCT,
     product: undefined,
+    cart: undefined,
+    quantity: 0,
     addressIds: [],
-    isPdp: true,
     total: {
       label: '',
       amount: '',
+      currency: '',
     },
   };
 
-  protected initLocalCart(product: Product, quantity: number): LocalCart {
-    return {
-      quantity,
-      product,
-      addressIds: [],
-      isPdp: true,
-      total: {
-        amount: ((product.price?.value as number) * quantity).toString(),
-        label:
-          quantity > 1
-            ? `${product?.name as string} x ${quantity}`
-            : `${product?.name as string}`,
-      },
-    };
+  protected transactionDetails = this.initialTransactionDetails;
+
+  protected initTransactionDetails(
+    transactionInput: ApplePayTransactionInput
+  ): ApplePayTransactionDetails {
+    this.transactionDetails = { ...this.initialTransactionDetails };
+
+    if (transactionInput?.cart) {
+      this.transactionDetails = {
+        ...this.transactionDetails,
+        context: OpfQuickBuyLocation.CART,
+        cart: transactionInput.cart,
+        total: {
+          amount: `${transactionInput.cart.totalPrice?.value}`,
+          label: `${transactionInput.cart.code}`,
```

Comment:
as discussed, the company name should be the total.label
https://developer.apple.com/documentation/apple_pay_on_the_web/applepaypaymentrequest/1916119-total
I'll create a separate ticket for it.